### DISCUSSION TITLE
Intel network card I219-LM (2) connectivity problem

### DISCUSSION THREAD

**Discussion by NashTOM** [ipxe/ipxe#695] [General]
Hello, 

I'm facing a connectivity problem when using Intel network card I219-LM (8086:15B7). 

I have compile the ipxe kernel with DEBUG=intel:3, you can see the traces that I get attached in the ticket. I have tried to monitor the traffic with wireshark and there are no packets transmitted. I have tested with 3fe683e kernel and the current ipxe master, getting the same problem.

I have the same problem launching ipxe with GRUB and without it. 

![ipxe1](https://user-images.githubusercontent.com/110162980/181471929-cf6b2544-2193-4a8f-9fb7-dfe505a49f5e.png)
![ipxe2](https://user-images.githubusercontent.com/110162980/181471935-c244fb59-14ee-41e2-a654-b8f85bea4ee6.png)
![ipxe3](https://user-images.githubusercontent.com/110162980/181471936-2318108f-ded0-4b7b-ba1e-06d8e5f1d7ab.png)
 
Can anyone give me any advice to solve the problem with this 'special' card? It seems that the undocumented bit in FEXTNVM11 is not working with this card. (PCI_ROM ( 0x8086, 0x15b7, "i219lm-2", "I219-LM (2)", INTEL_I219 ))

Thanks all!

Tom

**Comment by NiKiZe** [comment]
What is the link speed? (What does the switch show?) And have you tried with both 1gig and 100mbit and see the same result?

**Reply by NashTOM** [reply]
I have checked all the BIOS configuration but I cannot find anything related to the link speed, so i think it will be used auto configuration.
Is there any speed configuration supported by this card? I will try to get different switches in order to test different speeds.

**Reply by NiKiZe** [reply]
You have to change the device that the NIC is connected to. Generally the switch. But first step is to check if you have link, and what speed that is.

**Reply by NashTOM** [reply]
I have tested ifstat command and the link is up
![net_ifstat](https://user-images.githubusercontent.com/110162980/182097558-6f61e970-1c87-46b3-8c74-adad44470de1.png)

**Reply by stappersg** [reply]
(The _clickable version_ of the single _TXE_    http://ipxe.org/3c654003  and the twelve time _TXE_ http://ipxe.org/4c376035 )

Acknowledge on the `Link:up`.  Do note that _Transmission Error_ count is at 13.

Please report the speed  ( qouting NiKiZe: _"You have to change the device that the NIC is connected to. Generally the switch. But first step is to check if you have link, and what speed that is."_). 
So go to the management interface of the switch and check the line speed of the port that is connected to the I219-LM.

**Reply by NashTOM** [reply]
Sorry, the line speed of this port is 1000 Mb/s but i cannot change it. I'm trying to get another switch.

**Reply by NashTOM** [reply]
I have tried with a D-link 10/100 switch and I have the same error.

**Reply by NiKiZe** [reply]
Gig usually works while slower speeds might not.
Could you check if there is any firmware updates for your machine, maybe also disable the NIC ROM and run iPXE from usb. (Use intel.usb or similar build)

I have seen machines booting fine with EFI but fail in pcbios mode.
As a workaround you might be able to use undionly.kpxe.
